src/rule.rs

```diff
@@ -3,38 +3,70 @@ use std::sync::Arc;
 use derive_builder::Builder;
 use tree_sitter::{Node, Query};
 
-use crate::context::{Context, QueryMatchContext};
+use crate::{context::QueryMatchContext, Config};
 
-#[derive(Builder)]
-#[builder(setter(into))]
-pub struct Rule {
+#[derive(Clone)]
+pub struct RuleMeta {
     pub name: String,
-    #[builder(setter(custom))]
-    pub create: Arc<dyn Fn(&Context) -> Vec<RuleListener>>,
+    pub fixable: bool,
+}
+
+pub struct Rule {
+    pub meta: RuleMeta,
+    #[allow(clippy::type_complexity)]
+    pub create: Arc<dyn Fn(&Config) -> Vec<RuleListener>>,
 }
 
 impl Rule {
-    pub fn resolve(self, context: &Context) -> ResolvedRule<'_> {
-        let Rule { name, create } = self;
+    pub fn resolve(self, config: &Config) -> ResolvedRule<'_> {
+        let Rule { meta, create } = self;
 
         ResolvedRule::new(
-            name,
-            create(context)
+            meta,
+            create(config)
                 .into_iter()
-                .map(|rule_listener| rule_listener.resolve(context))
+                .map(|rule_listener| rule_listener.resolve(config))
                 .collect(),
         )
     }
 }
 
+#[derive(Clone, Default)]
+pub struct RuleBuilder {
+    name: Option<String>,
+    fixable: bool,
+    #[allow(clippy::type_complexity)]
+    create: Option<Arc<dyn Fn(&Config) -> Vec<RuleListener>>>,
+}
+
 impl RuleBuilder {
+    pub fn name(&mut self, name: impl Into<String>) -> &mut Self {
+        self.name = Some(name.into());
+        self
+    }
+
+    pub fn fixable(&mut self, fixable: bool) -> &mut Self {
+        self.fixable = fixable;
+        self
+    }
+
     pub fn create(
         &mut self,
-        callback: impl Fn(&Context) -> Vec<RuleListener> + 'static,
+        callback: impl Fn(&Config) -> Vec<RuleListener> + 'static,
     ) -> &mut Self {
         self.create = Some(Arc::new(callback));
         self
     }
+
+    pub fn build(&self) -> Result<Rule, ()> {
+        Ok(Rule {
+            meta: RuleMeta {
+                name: self.name.clone().ok_or(())?,
```

Comment:
Ah right because it's inside the `RuleBuilder` that I'm doing the "individual top-level arguments" (eg `name`) -> "nested inside `meta`" re-structuring, I just made `RuleBuilder` a manually-implemented builder struct

And so here this is a weird temporarily-not-giving-you-very-good-error-messages way to fail if you didn't supply expected "arguments" to the builder